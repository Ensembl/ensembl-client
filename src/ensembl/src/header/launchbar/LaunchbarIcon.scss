.launchbarIcon {
  height: 30px;
  width: 30px;
}
