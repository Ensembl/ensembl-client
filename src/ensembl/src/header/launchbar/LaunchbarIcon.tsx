import React, { FunctionComponent, memo, Props } from 'react';
import { NavLink } from 'react-router-dom';

import styles from './Launchbar.scss';

type LaunchbarIconProps = {
  app: string;
  description: string;
  icon: React.FunctionComponent<React.SVGProps<SVGSVGElement>> | string;
  enabled: boolean;
};

export const LaunchbarIcon: FunctionComponent<LaunchbarIconProps> = (
  props: LaunchbarIconProps
) => {
  const icon =
    typeof props.icon === 'string' ? (
      <img
        src={props.icon}
        className={styles.launchbarIcon}
        alt={props.description}
      />
    ) : (
      <props.icon />
    );

  return (
    <NavLink className={styles.launchbarButton} to={`/app/${props.app}`}>
      {icon}
    </NavLink>
  );
};

export default memo(LaunchbarIcon);
