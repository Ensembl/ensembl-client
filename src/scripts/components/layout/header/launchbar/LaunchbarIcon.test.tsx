import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  launchbarConfig,
  LaunchbarApp
} from '../../../../configs/launchbarConfig';

import LaunchbarIcon from './LaunchbarIcon';

describe('<LaunchbarIcon />', () => {
  const currentApp: string = 'about';
  const app: LaunchbarApp = Object.assign({}, launchbarConfig.about, {
    icon: {
      default: '/assets/img/launchbar/ensembl.svg',
      selected: '/assets/img/launchbar/ensembl-selected.svg'
    }
  });

  let gotoAppFn: (appName: string) => void;
  let wrapper: any;

  beforeEach(() => {
    gotoAppFn = jest.fn();

    wrapper = shallow(
      <LaunchbarIcon app={app} gotoApp={gotoAppFn} currentApp={currentApp} />
    );
  });

  test('displays selected icon if current app', () => {
    expect(wrapper.instance().getAppIcon()).toEqual(
      expect.stringContaining('selected')
    );
  });

  test('displays selected icon if not current app', () => {
    wrapper.setProps({ currentApp: 'browser' });

    expect(wrapper.instance().getAppIcon()).not.toEqual(
      expect.stringContaining('selected')
    );
  });

  test('should go to app on button click', () => {
    wrapper.find('button').simulate('click');

    expect(gotoAppFn).toHaveBeenCalled();
  });

  test('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
