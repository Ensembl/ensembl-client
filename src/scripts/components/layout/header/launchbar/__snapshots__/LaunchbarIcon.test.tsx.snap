// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchbarIcon /> renders correctly 1`] = `
<dt>
  <button
    onClick={[Function]}
  >
    <img
      alt="about ensembl"
      src="/assets/img/launchbar/ensembl-selected.svg"
      title="about ensembl"
    />
  </button>
</dt>
`;
